packages/preview-middleware-client/src/cpe/changes/flex-change.ts

```diff
@@ -0,0 +1,55 @@
+import type FlexCommand from 'sap/ui/rta/command/FlexCommand';
+import CommandFactory from 'sap/ui/rta/command/CommandFactory';
+import type { PropertyChange } from '@sap-ux-private/control-property-editor-common';
+import type { UI5AdaptationOptions } from '../types';
+
+/**
+ * Function to check a give value is a binding expression.
+ *
+ * @param value value to be checked.
+ * @returns boolean
+ */
+function isBindingExpression(value: string): boolean {
+    return value.includes('{') && value.includes('}');
+}
+
+/**
+ *
+ * @param options UI5 adaptation options
+ * @param change changed property of a control
+ */
+export async function applyChange(options: UI5AdaptationOptions, change: PropertyChange): Promise<void> {
+    const { rta } = options;
+    const modifiedControl = sap.ui.getCore().byId(change.controlId);
+    if (!modifiedControl) {
+        return;
+    }
+
+    const flexSettings = rta.getFlexSettings();
+
+    const changeType =
+        typeof change.value === 'string' && isBindingExpression(change.value) ? 'BindProperty' : 'Property';
+
+    const modifiedValue =
+        typeof change.value === 'string' && isBindingExpression(change.value)
+            ? {
+                  generator: flexSettings.generator,
+                  propertyName: change.propertyName,
+                  newBinding: change.value
+              }
+            : {
+                  generator: flexSettings.generator,
+                  propertyName: change.propertyName,
+                  newValue: change.value
+              };
```

Comment:
Can we refactor this to avoid the code duplication?

Reply:
Something like ...
```ts
const property = typeof change.value === 'string' && isBindingExpression(change.value) ? 'newBinding' : 'newValue';
const modifiedValue = {
   generator: flexSettings.generator,
   propertyName: change.propertyName,
   [property]: change.value
};
```

Reply:
Thanks.